src/common.c

```diff
@@ -1154,6 +1154,9 @@ ssize_t ofi_bsock_flush(struct ofi_bsock *bsock)
 				   &bsock->sq.data[bsock->sq.head],
 				   avail, MSG_NOSIGNAL, bsock);
 	if (ret < 0) {
+		if (ret == -FI_EIOURING_PREP)
```

Comment:
Is this equivalent to EINPROGRESS?  That is, an async operation was started?

Reply:
Yes, it's strictly equivalent to `EINPROGRESS`. I have used a different error code because otherwise the caller cannot differentiate the underlying implementation (sockets vs. io_uring) and cannot decide what to do next.

Reply:
EINPROGRESS currently implies that we're using async sockets.  My preference is either to use this return code in all cases and use another method to determine how to respond to the async request; OR define 2 provider specific error codes, for both io_uring and async sockets, rather than using a standard code in one case and not the other (e.g. EINPROGRESS_ASYNC vs EINPROGRESS_URING? -- not that I'm thrilled with either name).

We may be able to setup an async_handler() based on which type of async method we're using.  Using this method may allow just using EINPROGRESS and responding to that return accordingly.  But having separate return codes may make the code easier to follow, since we wouldn't need to follow function pointers.

Reply:
I like the suggestion of using 2  error codes for both io_uring and async sockets. 
One problem is that the functions used by `sockapi` are defined in common code (`src/common.c` and `src/iouring.c`): I'm not sure how we can reference a provider specific error code in the common code.

Reply:
Hmm... efa mostly starts their error codes above 4k.  They have some lower values, but the header indicates that they're trying to avoid rdma-core error codes.  I guess we could add internal error codes in ofi.h or wherever sockapi lands starting at 512.  We can always change internal error codes, but that leaves plenty of space after FI_ERRNO_MAX.